### PR TITLE
Move project and build configuration to `pyproject.toml` and use `setuptools-scm` for generating package version from Git tags

### PR DESCRIPTION
**Pull request by hoechenberger**
I cannot move the flake8 config to pyproject.toml because this package doesn't support this.

In the medium term, we should probably move to ruff anyway.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -115,9 +115,9 @@ jobs:
         pyvista: false
     # Have to install any extra `main` MNE deps here because we use get_testing_version.sh below
     - name: Install autoreject
-      run: |
+      run: |g
```
Comment (hoechenberger) [inline .github/workflows/test.yml:118]:
```suggestion
      run:
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -84,7 +84,7 @@ jobs:
     - uses: actions/checkout@v4
     - name: Test extras install
       run: |
-        python -m pip install .[test,doc]
+        python -m pip install ".[test,doc]"
```
Comment (hoechenberger) [inline .github/workflows/test.yml:87]:
```suggestion
        python -m pip install .[test,doc]
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -115,7 +115,7 @@ jobs:
         pyvista: false
     # Have to install any extra `main` MNE deps here because we use get_testing_version.sh below
     - name: Install autoreject
-      run: |
+      run:
```
Comment (hoechenberger) [inline .github/workflows/test.yml:118]:
```suggestion
      run: |
```

**Comment by hoechenberger** [comment]
FNIRS test failure seems unrelated…

**Comment by jasmainak** [comment]
@hoechenberger looks good but seems you created a branch on upstream instead of your own fork.

Is it ready to merge?

**Comment by hoechenberger** [comment]
Oh I'm sorry, this was a mistake!

But from my end this is ready to be merged. Let's wait for @larsoner go take a look. There is no time pressure here!

**Review by sappelhoff — COMMENTED** [review]
I think we should move the flake8 config to pyproject.toml in this PR, and simply use this package to make it work: https://pypi.org/project/Flake8-pyproject/

(until switching over to ruff)

**Comment by hoechenberger** [comment]
> I think we should move the flake8 config to pyproject.toml in this PR, and simply use this package to make it work: https://pypi.org/project/Flake8-pyproject/

I don't think it's worth the time and effort, since we'll likely switch to ruff soon

**Comment by sappelhoff** [comment]
Thanks @hoechenberger

**Pull request merged**